src/ScssAssetConverter.php

```diff
@@ -26,7 +26,7 @@ class ScssAssetConverter extends Component implements AssetConverterInterface
      */
     public $forceConvert = false;
 
-    private $compiler;
+    protected $compiler;
```

Comment:
Making the compiler injectable should be done in a different way, since currently the constructor instantiates it always, which would be wasteful if an extension class replaces it anyway. There are multiple ways:

1. The caller should put a custom Compiler extension in the DI container so that `Yii::createObject(Compiler::class)` in `init()` uses the new one automatically. The class member can stay private so no modification is needed.
2. Make use of a member `public $compilerClass = Compiler::class;` (string). Then instead of saying `Yii::createObject(Compiler::class)` in `init()`, it should say `Yii::createObject($this->compilerClass)`. This is more idiomatic since it is seen in various parts of the framework, and the desired compiler class can be overridden by giving it to the options in the constructor.